File: back-end/Mercury.Engine/Mercury.Engine.API/Services/UserService.cs
using Grpc.Core;
using System.Threading.Tasks;
using Mercury.Engine.API.DbContext.UnitOfWork;
using Mercury.Engine.API.Services.GrpcGenerated;
using Mercury.Engine.API.Models.GrpcGenerated;

namespace Mercury.Engine.API.Services
{
    public class UserService : Users.UsersBase
    {
        private readonly IUnitOfWork _unitOfWork;

        public UserService(IUnitOfWork unitOfWork) => _unitOfWork = unitOfWork;

        public override async Task<GetUserByTagReply> GetUserByTag(GetUserByTagRequest request, ServerCallContext context)
        {
            var user = await _unitOfWork.TbUserRepository.GetUserByTag(request.UserTag).ConfigureAwait(false);

            return new GetUserByTagReply
            {
                User = new User
                {
                    UserId = user.IdUser,
                    UserTag = user.TxUserTag,
                    UserEmail = user.TxEmail
                }
            };
        }
    }
}

File: front-end/mercury/lib/services/user_service.dart
import 'package:mercury/services/service_base.dart';

import 'gen/services/users.pb.dart';
import 'gen/services/users.pbgrpc.dart';

class UserService extends ServiceBase {
  Future<GetUserByTagReply> getUserByTag(String tag) async {
    final clientChannel = await setup.clientChannel;
    final client = UsersClient(clientChannel);

    var request = GetUserByTagRequest()..userTag = tag;

    GetUserByTagReply reply;
    try {
      reply = await client.getUserByTag(request);
    } catch (e) {
      print(e.toString());
    }
    // await clientChannel.shutdown();
    return reply;
  }
}
